db/design_doc.go

```diff
@@ -612,13 +613,30 @@ func WaitForViews(bucket base.Bucket) error {
 func waitForViewIndexing(bucket base.Bucket, ddocName string, viewName string) error {
 	var vres interface{}
 	opts := map[string]interface{}{"stale": false, "key": fmt.Sprintf("view_%s_ready_check", viewName), "limit": 1}
+
+	// Not using standard retry loop here, because we want to retry indefinitely on view timeout (since view indexing could potentially take hours), and
+	// we don't need to sleep between attempts.  Using manual exponential backoff retry processing for non-timeout related errors, waits up to ~5 min
```

Comment:
I don't think the `waits up to ~5 min` comment is correct (based on current values for backoff).

You could start retrySleep at 1000ms to make the longest individual wait 5 minutes

attempt | retrySleep (ms) | seconds | minutes |   |   | retrySleep (ms) | seconds | minutes
-- | -- | -- | -- | -- | -- | -- | -- | --
1 | 100.00 | 0.10 | 0.00 |   |   | 1000.00 | 1.00 | 0.02
2 | 150.00 | 0.15 | 0.00 |   |   | 1500.00 | 1.50 | 0.03
3 | 225.00 | 0.23 | 0.00 |   |   | 2250.00 | 2.25 | 0.04
4 | 337.50 | 0.34 | 0.01 |   |   | 3375.00 | 3.38 | 0.06
5 | 506.25 | 0.51 | 0.01 |   |   | 5062.50 | 5.06 | 0.08
6 | 759.38 | 0.76 | 0.01 |   |   | 7593.75 | 7.59 | 0.13
7 | 1139.06 | 1.14 | 0.02 |   |   | 11390.63 | 11.39 | 0.19
8 | 1708.59 | 1.71 | 0.03 |   |   | 17085.94 | 17.09 | 0.28
9 | 2562.89 | 2.56 | 0.04 |   |   | 25628.91 | 25.63 | 0.43
10 | 3844.34 | 3.84 | 0.06 |   |   | 38443.36 | 38.44 | 0.64
11 | 5766.50 | 5.77 | 0.10 |   |   | 57665.04 | 57.67 | 0.96
12 | 8649.76 | 8.65 | 0.14 |   |   | 86497.56 | 86.50 | 1.44
13 | 12974.63 | 12.97 | 0.22 |   |   | 129746.34 | 129.75 | 2.16
14 | 19461.95 | 19.46 | 0.32 |   |   | 194619.51 | 194.62 | 3.24
15 | 29192.93 | 29.19 | 0.49 |   |   | 291929.26 | 291.93 | 4.87
| | | **Sum:** | **1.46** |   |   |  | **Sum:** | **14.56** |

Reply:
Thanks for catching - I think that was left over from when I was doing 2x instead of 1.5x.  Will tweak that.